### DISCUSSION TITLE
match_weights_chart all fields have grey

### DISCUSSION THREAD

**Discussion by theimanph** [moj-analytical-services/splink#1949] [Q&A]
Hi, 

When I create the match_weights chart I get a graph where everything is grey. See attached graph. 
[grey_graph.pptx](https://github.com/moj-analytical-services/splink/files/14214048/grey_graph.pptx)
My code is below. Am I doing something obviously stupid? Thank you!

Sincerely,

tom

blocking_rules = [
        block_on("lastname_dm"),
        block_on("PER_LastName"),
        block_on("new_address"),
        #block_on("full_name"),
        block_on("firstname_dm"),
        block_on("PER_FirstName"),
        block_on("PER_DOB"),
        block_on("zip_valid"),
        block_on("PER_HomePhone"),
        block_on("PER_Email"),
        block_on("PER_CellularPhoneOrPager"),
        block_on("middlename_dm")
]

import splink.duckdb.comparison_template_library as ctl
from splink.duckdb.linker import DuckDBLinker
import splink.duckdb.comparison_library as cl

import splink.duckdb.comparison_library as cl
import splink.duckdb.comparison_template_library as ctl

settings = {
    "unique_id_column_name": "id",
    "link_type": "dedupe_only",
    "blocking_rules_to_generate_predictions": blocking_rules,
    "comparisons": [
        ctl.name_comparison("PER_FirstName"),
        ctl.name_comparison("PER_LastName"),
        ctl.name_comparison("full_name"),
        ctl.email_comparison("PER_Email", include_username_fuzzy_level=False),
        ctl.forename_surname_comparison("PER_FirstName","PER_LastName"),
        cl.levenshtein_at_thresholds("PER_SSN", [2]),
        cl.exact_match("new_address", term_frequency_adjustments=True),
        ctl.date_comparison("PER_DOB"),
    ],
    "retain_intermediate_calculation_columns": True
}

linker = DuckDBLinker(cleaned_df9_subset, settings, set_up_basic_logging=False)
deterministic_rules = [
    "l.PER_SSN = r.PER_SSN and l.PER_DOB = r.PER_DOB",
    "l.PER_FirstName = r.PER_FirstName and l.PER_DOB = r.PER_DOB",
    "l.PER_LastName = r.PER_LastName and l.PER_DOB = r.PER_DOB",
]

linker.estimate_probability_two_random_records_match(deterministic_rules, recall=0.6)
linker.estimate_u_using_random_sampling(max_pairs=1e8)
linker.estimate_m_from_label_column("PER_SSN")
linker.estimate_m_from_label_column("PER_CellularPhoneOrPager")

This gives me a complete model and I can make the graph..

 

<img width="752" alt="image" src="https://github.com/moj-analytical-services/splink/assets/153543409/1da5b900-51a5-45ef-946a-323a9638b221">

**Comment by RobinL** [comment]
It looks like you're including the same information in multiple comparisons - each column (e.g. first name) should be included in one `Comparison` only
